src/main/groovy/life/qbic/portal/sampletracking/app/samples/list/ModifySampleList.groovy

```diff
@@ -0,0 +1,48 @@
+package life.qbic.portal.sampletracking.app.samples.list
+
+import groovy.util.logging.Log4j2
+import life.qbic.datamodel.samples.Sample
+
+@Log4j2
+class ModifySampleList implements ModifySampleListInput {
+
+    ModifySampleListOutput output
+
+    ModifySampleList() {
+
+    }
+
+    ModifySampleList(ModifySampleListOutput output) {
+        this()
+        this.injectSampleListOutput(output)
+    }
+
+    @Override
+    void selectSamples(Sample... samples) {
+        this.output.addSamples(samples)
+    }
+
+    @Override
+    void deselectSamples(Sample... samples) {
+        this.output.removeSamples(samples)
+    }
+
+    @Override
+    void clearSelection() {
+        this.output.clearSamples()
+    }
+
+
+    /**
+     * This method is an alternative to do dependency injection.
+     * It is used to avoid circular dependencies.
+     * The method does set the sample list output only once and does nothing if it is already set.
+     */
+    void injectSampleListOutput(ModifySampleListOutput sampleListOutput) {
```

Comment:
Do we still need this method? I thought that we can now inject the output directly on construction?

Reply:
No we do not need it and it should be removed. I overlooked this yesterday.